calyx/src/passes/compile_ref.rs

```diff
@@ -0,0 +1,114 @@
+use super::dump_ports;
+use crate::errors::CalyxResult;
+use crate::ir::traversal::{
+    Action, ConstructVisitor, Named, VisResult, Visitor,
+};
+use crate::ir::Attributes;
+use crate::ir::WRC;
+use crate::ir::{self, LibrarySignatures, RRC};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+pub struct CompileRef {
```

Comment:
This struct needs documentation string describing how it compiles `ref` cells and `invoke` statements with `ref` cells.